src/version.ts

```diff
@@ -28,11 +28,11 @@ export class MiraVersion {
      * Checks a particular dependency CDK version.
      */
     /* eslint-disable-next-line */
-    checkApplicationDependencyCDKVersion (pkg: any, dep: string, autoFix = true): boolean {
+    checkApplicationDependencyCDKVersion (pkg: any, pkgDeps: object, dep: string, autoFix = true): boolean {
       if (!dep.startsWith('@aws-cdk') && !dep.startsWith('aws-cdk')) {
         return false
       }
-      const cdkVersion = pkg.dependencies[dep]
+      const cdkVersion = pkgDeps[dep]
```

Comment:
I think `cdkVersion` can be undefined which will cause a problem in the `checkDependency` function as the first argument must be a string.

Reply:
I traced through and this should throw an error which would be expected.  It may be less than clear though that this is the intended behavior.  What do you think?  Do you want me to try to make  the code clearer?